azurerm/data_source_redis_cache.go

```diff
@@ -0,0 +1,266 @@
+package azurerm
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmRedisCache() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmRedisCacheRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": locationForDataSourceSchema(),
+
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
+
+			"zones": singleZonesSchema(),
```

Comment:
since this is readonly, this'll need to be:

```suggestion
			"zones": zonesSchemaComputed(),
```

(as otherwise it's possible to specify a value for this field)